### ISSUE TITLE
How is PySimpleGUI working out for you??

### ISSUE THREAD

**Issue by MikeTheWatchGuy** [neilmaldy/Psirt#1]
Was shocked to see so much code using PySimpleGUI!  

Yikes!  It's a little scary to see.  I don't use or rarely use some of the features you're using like multiple-selection listboxes.  

Also see you're using the Debug Print.  It's another kinda spooky area since it's a persistent window and hasn't been used very much.

Checking in to make sure it's all performing OK for you.  Let me know if I can help.

**Comment by neilmaldy** [comment]
Great job on pysimplegui, I was able to add a fully functional gui to a cli
app in less than a day!

I'm curious, how did you find the github repro so quickly?  I just started
the gui work today.

**Comment by MikeTheWatchGuy** [comment]
I've been on the lookout, that's how!
I get so little feedback on how people are using it, or not using it, that I'm actively out looking for uses of it.  It's the only way at the moment to measure if it's "successful" at it's purpose.  In this case, you've answered my question perfectly:
    I was able to add a fully functional gui to a cli app in less than a day!

THIS is why I wrote and released the package.  I have been adding GUI after GUI to my own stuff and seem to be able to effortlessly make these things, but I haven't known if it's going to work well for others.

I expected to see some code to collect filenames, etc, but then was shocked to see more gui code as I scrolled down the page!  

I hope it continues to work out well for you.  If you do have trouble, you know I'm reachable :-)

BTW.... just added this one-line GUI to my Cookbook so that perhaps others will use it to bolt-on a quick file-browser to command line programs.  You've already done much more than this clearly!

```
import PySimpleGUI as sg
import sys

'''
Quickly add a GUI to your script!

This simple script shows a 1-line-GUI addition to a typical Python command line script.

Previously this script accepted 1 parameter on the command line.  When executed, that
parameter is read into the variable fname.

The 1-line-GUI shows a form that allows the user to browse to find the filename. The GUI
stores the result in the variable fname, just like the command line parsing did.
'''

if len(sys.argv) == 1:
    button, (fname,) = sg.FlexForm('My Script').LayoutAndRead([[sg.T('Document to open')],
                                                           [sg.In(), sg.FileBrowse()],
                                                           [sg.Open(), sg.Cancel()]])
else:
    fname = sys.argv[1]

if not fname:
    sg.MsgBox("Cancel", "No filename supplied")
    raise SystemExit("Cancelling: no filename supplied")

```

**Comment by neilmaldy** [comment]
The quick file browser is pretty slick.  I gave your project a thumbs up on
vinta/awesome-python - again, great job!

**Comment by MikeTheWatchGuy** [comment]
Huh?  Quick file browser?  PNG viewer or PDF viewer maybe?  
Vinta / awesome python?  Lots of stuff that sounds great but I don't recognize it.

**Comment by MikeTheWatchGuy** [comment]
Oh oh... you mean the 1-line file browser :-)  Sorry, I'm slow evidently.

And I see that I added PySimpleGUI to the Vinta list.... but it doesn't appear to be merged.   Not sure if I did something incorrectly there or not.  I'm surprised you found it there.

**Comment by MikeTheWatchGuy** [comment]
Can you click the smiley and choose thumbs up on this page?
https://github.com/vinta/awesome-python/pull/1100

I think that's the one where you vote.

**Comment by neilmaldy** [comment]
Yes, did that.  By the way, that's also how I heard about your project back at the beginning of Aug.

**Comment by MikeTheWatchGuy** [comment]
Another oops on my part. 
I clearly mis-parsed several of your message. Sorry about that.

I'm shocked you found it this way.  You must stay on the CUTTING edge to be going through items that have not yet been approved.

If you've been following for the past month, then you've likely seen a lot of new features get dumped in. I'm kinda surprised by how much got in there... and how relatively easy it was.  The most difficult feature was Columns because it made forms recursive.  Other than that, many feature requests were 2 and 3 lines.

Dictionary Return Values - The best new feature was recommended on a forum.  Wish I could take credit for that one. I use them often now.

Thanks for your support. It's really appreciated.

**Comment by MikeTheWatchGuy** [comment]
Released 3.1.0 to PyPI today.
It has menus, almost the last of the big features to add.

The missing one at this point is TreeView which can display folders or tables. I don't understand enough about it to get how those 2 things belong together.

Thought maybe Menus could be of interest to your project.
